src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,43 +2,40 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
 
 public class ProductBasket {
 
-    private final Product[] products = new Product[5];
+    LinkedList<Product> products = new LinkedList<>();
```

Comment:
поля должны быть приватные всегда. Также тип ссылки должен быть максимально общим (см. полиморфизм)
```suggestion
    private List<Product> products = new LinkedList<>();
```